### PR TITLE
Add intermediate clairvoyance level. 

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
Implementation of what @kevingranade said to do in #21721

I had to make a new PR since I messed up big time and had to start over. The two other clairvoyance levels were left alone and the new level was added to some vanilla artifacts to both make use of the new level and as a test of sorts.

### CODE REVIEW

**Comment by kevingranade** [comment]
I'm not sure what was unclear about, "Clairvoyance lets you see what's on the other side of walls, and that's it." and "use (the new) effect in the mod.".

Add a use of the new ability in a debug item or a mod, not in vanilla game content.

**Comment by Noctifer-de-Mortem** [comment]
Understood, will fix asap.

File: src/character.cpp [hunk 1]
```diff
@@ -504,6 +504,14 @@ void Character::recalc_sight_limits()
         vision_mode_cache.set( VISION_CLAIRVOYANCE_SUPER );
     }
 
+    if( has_artifact_with( AEP_SUPER_CLAIRVOYANCE ) ) {
```
Comment (BevapDin) [inline src/character.cpp]: Why has this block been duplicated? The very same code already exists above.

File: src/enums.h [hunk 2]
```diff
@@ -114,6 +114,7 @@ enum art_effect_passive : int {
     AEP_SNAKES, // Summons friendly snakes when you're hit
     AEP_INVISIBLE, // Makes you invisible
     AEP_CLAIRVOYANCE, // See through walls
+    AEP_CLAIRVOYANCE_PLUS, // See through walls to a larger distance
```
Comment (Coolthulhu) [inline src/enums.h]: Note that adding anything to a hardcoded enum is highly likely to scramble existing saves that use this enum. At the very least old saves from 0.C stable, which are still supposed to be handled.

Comment (Noctifer-de-Mortem) [reply src/enums.h]: Would placing it at the end of the list be enough?

Comment (BevapDin) [reply src/enums.h]: Probably, put it right before `NUM_AEPS`.

**Pull request merged**